Dart/src/com/jetbrains/lang/dart/ide/errorTreeView/DartProblemsFilter.java

```diff
@@ -109,6 +108,14 @@ public boolean include(@NotNull final DartProblem problem) {
       return false;
     }
 
+    if (myFileFilterMode == FileFilterMode.Directory && myCurrentFile != null) {
```

Comment:
I've rewrote it like this. Let me know in case of objections.
```
    if (myFileFilterMode == FileFilterMode.Directory) {
      if (myCurrentFile == null) return false;
      ...
    }
```